bHaptics Integration 1.20.4

finaltransmit opened the pull request:
Updated my old bHaptics integration to 1.20.4

finaltransmit committed:
bHaptics Integration

fayer3 commented:
would you be willing to fix the conflicts and change the math lib to joml, or should I do that?

fayer3 committed:
use joml and some fixes

fayer3 committed:
- add damage direction packet
- improve damage direction detection
- add top/bottom/all around damage haptics

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.21.4' into bhaptics-rebase

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.21.4' into Multiloader-1.20.4

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.21.4' into Multiloader-1.20.4

fayer3 committed:
don't use our bHaptics integration if the official one is loaded
correctly disconnect bHaptics when disabling VR

fayer3 committed:
fix crash with bHaptics not connected and explosions/eating

fayer3 committed:
this shouldn't have been there

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.21.4' into Multiloader-1.20.4

fayer3 committed:
better add the VR check as well

fayer3 committed:
fix wrong Charsets import

fayer3 committed:
Merge remote-tracking branch 'remotes/origin/Multiloader-1.21.4' into Multiloader-1.20.4

fayer3 committed:
make enums capitalized
